scripts/2025-10-spike-ingredient-density/llm.py

```diff
@@ -47,6 +47,14 @@ def call_llm(self, prompt: str, tool: dict, temperature: float = 0) -> dict:
 
     # Decode the response body.
     model_response = json.loads(response["body"].read())
-    obj_response = model_response["content"][0]["input"]
+    obj_response = None
+    for content in model_response["content"]:
+      if content["type"] == "tool_use":
+        obj_response = content["input"]
+      if content["type"] == "text":
+        print(f"LLM TEXT: {content['text']}")
+
+    if obj_response is None:
+      raise ValueError("No tool_use response from LLM")
```

Comment:
Just curious to know - are we stopping execution with this error and not handling it ya? so is their any re-try or something we would be doing? or just it is based on purely incoming request so no retry needed?

Reply:
no retry whatsoever, it's non production code so if it crashes, it crashes!

That being said it run continuously for an hour without encountering any error.